qemu/tests/migration_after_vm_paused.py

```diff
@@ -0,0 +1,227 @@
+            vm.migrate(mig_timeout, mig_protocol, mig_cancel_delay,
+                       offline, check,
+                       migration_exec_cmd_src=mig_exec_cmd_src,
+                       migration_exec_cmd_dst=mig_exec_cmd_dst, env=env)
+
+        # Set deamon thread action to stop after migrate
+        params["action"] = "stop"
+
+        # Cont guest
+        if vm.is_paused():
+            vm.resume()
+
+        # Log into the guest again
+        logging.info("Logging into guest after migration...")
+        session2 = vm.wait_for_login(timeout=360)
+        logging.info("Logged in after migration")
+
+        # Make sure the background process is still running
+        error.context("Checking the background command in the guest "
+                      "post migration", logging.info)
+        session2.cmd(check_command, timeout=30)
+
+        # Get the output of migration_test_command
+        output = session2.cmd_output(test_command)
+
+        # Compare output to reference output
+        if output != reference_output:
+            logging.info("Command output before migration differs from "
+                         "command output after migration")
+            logging.info("Command: %s", test_command)
+            logging.info("Output before:" +
+                         utils_misc.format_str_for_message(reference_output))
+            logging.info("Output after:" +
+                         utils_misc.format_str_for_message(output))
+            raise error.TestFail("Command '%s' produced different output "
+                                 "before and after migration" % test_command)
+        # Kill the background process
+        if session2 and session2.is_alive():
+            bg_kill_cmd = params.get("migration_bg_kill_command", None)
+            ignore_status = params.get("migration_bg_kill_ignore_status", 1)
+            if bg_kill_cmd is not None:
+                try:
+                    session2.cmd(bg_kill_cmd)
+                except aexpect.ShellCmdError, details:
+                    # If the migration_bg_kill_command rc differs from
+                    # ignore_status, it means the migration_bg_command is
+                    # no longer alive. Let's ignore the failure here if
+                    # that is the case.
+                    if not int(details.status) == int(ignore_status):
+                        raise
+                except aexpect.ShellTimeoutError:
+                    logging.debug("Remote session not responsive.")
+        if deamon_thread is not None:
+            # Set deamon thread action to stop after migrate
+            params["action"] = "stop"
+            deamon_thread.join()
+
+        # reboot and shutdown the guest
+        vm.reboot()
+        vm.graceful_shutdown(timeout=60)
+    else:
+        # Just migrate on a living guest OS
+        raise error.TestFail("The guest is not alive,\
+this test must on a living guest OS.")
```

Comment:
I know why you use '\',  but you don't understand what I told you.
 "\" looks ugly.  how about below?
raise error.TestFail("The guest is not alive"
                                 " this test must on a living guest OS.")